scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -823,8 +821,16 @@ class FormatOps(val tree: Tree, val initStyle: ScalafmtConfig) {
       if (isBracket) close // If we can fit the type params, make it so
       else lastParen // If we can fit all in one block, make it so
 
+    val arity = valueParamsOwner match {
```

Comment:
I'm cheating here. I'd prefer to make this decision for each parameter group (in the case of functions) but I couldn't figure out how.

Reply:
Hmm, not sure I follow. Can you elaborate?

Reply:
Hah, my bad. I'm more confused than I though it seems. Given this code I would have assumed that the following would've triggered a reformat:

```scala
object A {
  def hello(how: String)(are: String)(you: String) = how + are + you
}
```

as it would hit `case d: Decl.Def` and `d.paramss.foldLeft(0)(_ + _.size)` would return `3`.

I extended the test and it does the right thing, I'm just not sure why.

Reply:
If I add `Defn.Def` to the match case it does indeed end up formatting it as 

```scala
object A {
  def hello(
    how: String
  )(are: String
  )(you: String
  ) = how + are + you
}
```

Which isn't quite what I'd want 😅

Reply:
I came up with a solution in 4a034ce